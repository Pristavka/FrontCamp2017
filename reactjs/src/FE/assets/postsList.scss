.title {
  font-family: 'Oswald Regular',Arial,sans-serif;
  text-transform: uppercase;
  letter-spacing: .15px;
  font-size: 16px;
  text-align: center;
  color: #1a9cb0;
}
