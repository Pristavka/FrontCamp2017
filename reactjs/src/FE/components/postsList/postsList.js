import React from 'react';
import { connect } from 'react-redux';
import styles from '../../assets/postsList.scss';
import Post from '../post/post';
import Sorting from '../sorting/sorting';
import config from '../../../configs/config';

const renderPosts = (posts) => (
  <React.Fragment>
    {posts.map(post => (<Post post={post} key={post._id} />))}
  </React.Fragment>
)

export class PostsList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { posts: props.posts }
  }

  componentDidMount() {
    if(!this.state.posts) this.props.fetchAllPosts(config.getAllPostsURL)
  }

  sortPosts = sortedPosts => this.setState(sortedPosts => ({ posts: sortedPosts }))

  render() {
    const posts = renderPosts(this.state.posts);
    return (
      <React.Fragment>
        <Sorting
          posts={this.state.posts}
          sortPosts={this.sortPosts}
        />
        <h1 className={styles.title}>This is the List fo all Blogs</h1>
        {posts}
      </React.Fragment>
    )
  }
};

export default connect(
  state => ({ posts: state.posts }),
  dispatch => ({ fetchAllPosts })
)(PostsList);
