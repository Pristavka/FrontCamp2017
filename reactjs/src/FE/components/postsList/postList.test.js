import React from 'react';
import { PostsList } from './postsList';

it('Should render a PostsList', () => {

  const props = {
    fetchAllPosts: jest.fn(),
    posts: [{
      '_id': 12354347,
      'author': 'Siarhei',
      'text': 'Hello World!'
    }]
  }

  const wrapper = shallow(<PostsList {...props}/>);
  expect(wrapper).toMatchSnapshot();
});

