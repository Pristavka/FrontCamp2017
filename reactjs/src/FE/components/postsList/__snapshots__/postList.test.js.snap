// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a PostsList 1`] = `
<React.Fragment>
  <Sorting
    posts={
      Array [
        Object {
          "_id": 12354347,
          "author": "Siarhei",
          "text": "Hello World!",
        },
      ]
    }
    sortPosts={[Function]}
  />
  <h1
    className="title"
  >
    This is the List fo all Blogs
  </h1>
  <React.Fragment>
    <Post
      key="12354347"
      post={
        Object {
          "_id": 12354347,
          "author": "Siarhei",
          "text": "Hello World!",
        }
      }
    />
  </React.Fragment>
</React.Fragment>
`;
